task9/index.js

```diff
@@ -0,0 +1,51 @@
+const obj = [
+    {to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"}, 
+    {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}
+]
+
+const div = document.getElementById("js-parent");
+
+const loading = () => {
+    const loadingPlace = document.createElement("div");
+    const gif = document.createElement('img');
+    loadingPlace.id = "loadingPlace";
+    gif.src = "loading-circle.gif";
+    div.appendChild(loadingPlace).appendChild(gif);
+};
+
+const removeLoading = () => {
+    document.getElementById("loadingPlace").remove();
+};
+
+const getData = () => {
+    return new Promise((resolve) => {
+        setTimeout(() => {
+            resolve(obj);
+        },2000);
+    });    
+};
+
+const objShow = async () => {
+    const value = await getData();
+    const ul = document.createElement('ul');
+    const fragment = document.createDocumentFragment();
+
+    removeLoading();
+    
+    value.forEach(item => {
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+
+        a.textContent = item.text;
+        a.href = item.to;
+        img.src = item.img;
+        img.alt = item.alt;
+
+        fragment.appendChild(li).appendChild(a).appendChild(img);
+    })
+    div.appendChild(ul).appendChild(fragment);
+};
+
+loading();
+objShow();
```

Comment:
How about running loading at the very beginning of objShow?

Reply:
Thanks for the comment!
I've changed where the loading runs, so it's easier to see.
f7eaca8